boards: nordic: nrf7002dk: Remove NS platforms

krish2718 opened the pull request:
NS platforms are not supported due to lacking crypto support (only supported in NCS), so, remove them from upstream.

Fixes #78942

PerMac reviewed (changes requested):
I believe it is wrong to have a platform in tree which is completely untestable. Also, this is a hacky workaround, not a proper use of tags.

PerMac commented:
"NS platforms are not supported due to lacking crypto support (only supported in NCS), so, exclude them from testing." If they are not supported, than they shouldn't be tree, no? If you say they need NCS, then why they are not placed in NCS?

nordicjm commented:
I agree, if the boards here do not work, remove them and put them downstream, they can be brought back when they are working.

krish2718 commented:
> Also, this is a hacky workaround, not a proper use of tags.

Yes, as mentioned in the description.

ok, I will move the board variant downstream.

krish2718 committed:
boards: nordic: nrf7002dk: Remove NS variants

The NS variants need crypto changes from NCS, so, remove it from
upstream till their support is upstreamed.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 commented:
@PerMac  please review, as this is gating nRF70 PRs want to get this merged ASAP.

anangl commented on boards/nordic/nrf7002dk/board.cmake:
```diff
@@ -1,15 +1,8 @@
 
-if(CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP_NS OR
-   CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP_NRF7001_NS)
-  set(TFM_PUBLIC_KEY_FORMAT "full")
```
Also the following is only related to non-secure targets so it's unneeded upstream: https://github.com/zephyrproject-rtos/zephyr/blob/733225e4a0322575006cca97b0d1d65f0664843c/boards/nordic/nrf7002dk/board.cmake#L9-L11

anangl commented on boards/nordic/nrf7002dk/Kconfig.defconfig:
```diff
@@ -4,15 +4,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if BOARD_NRF7002DK_NRF5340_CPUAPP || \
-	BOARD_NRF7002DK_NRF5340_CPUAPP_NS || \
```
Everything in this file is related to TF-M and does not make much sense if non-secure targets are not present. It'd be better to remove this file completely.

anangl commented on boards/nordic/nrf7002dk/Kconfig.defconfig:
Everything in this file is related to combined secure/non-secure builds so it does not make much sense to keep it if non-secure targets are not present.

krish2718 replied:
Yeah, I left these in intentionally to make it easier to reinstate but I can remove them as well.
